packages/react-app/src/components/history/HistoryItem.jsx

```diff
@@ -135,7 +136,7 @@ export const HistoryItem = ({
       } catch (executeError) {
         logError({ executeError, chainId: providerChainId, message });
         if (executeError && executeError.message) {
-          showError(executeError.message);
+          handleClaimError(toToken);
```

Comment:
We need to show this modal only in case the error is thrown when the claim was already executed. 
For example, with this current implementation, even when the user rejects the Metamask popup for transaction confirmation this modal is shown. 

Ideally this modal should be shown only in case the execution has already happened.

Reply:
Consider changing the logic in lib/amb.js:40 method executeSignatures 
to first check if the transfer was executed by calling the method messageCallStatus on the amb contract on foreign side.
Ref: https://github.com/poanetwork/tokenbridge-contracts/blob/master/contracts/upgradeable_contracts/arbitrary_message/MessageProcessor.sol#L12

If messageCallStatus returns true, then we can throw a custom error that says message was already executed.
And we can handle that error only to show the modal you've created. For all other cases, we must default to showing the toast as before.

Reply:
I am a lil stuck on finding what exactly is _messageId ..